data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/module/DataPrepperJakartaValidationModule.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.schemas.module;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.victools.jsonschema.generator.MemberScope;
+import com.github.victools.jsonschema.module.jakarta.validation.JakartaValidationModule;
+import com.github.victools.jsonschema.module.jakarta.validation.JakartaValidationOption;
+
+/**
+ * Custom {@link JakartaValidationModule} which overrides the default behavior of {@link JakartaValidationModule}
+ * for Data Prepper.
+ * It considers the {@link JsonProperty} annotation as well as the Jakarta validations.
+ */
+class DataPrepperJakartaValidationModule extends JakartaValidationModule {
+    public DataPrepperJakartaValidationModule(final JakartaValidationOption... options) {
+        super(options);
+    }
+
+    @Override
+    protected boolean isRequired(final MemberScope<?, ?> member) {
+        final JsonProperty jsonPropertyAnnotation = member.getAnnotationConsideringFieldAndGetter(JsonProperty.class);
+        if(jsonPropertyAnnotation != null) {
+            if(jsonPropertyAnnotation.defaultValue() != null && !jsonPropertyAnnotation.defaultValue().isEmpty()) {
```

Comment:
Do we have any cases where null value is allowed but defaultValue is not null?

Reply:
Technically you cannot provide `null` to the the annotation's `defaultValue` field. So this should never be reached. I added the `!= null` as a defensive programming approach.

Reply:
For a more general point-of-view, anytime that a `defaultValue` is provided, regardless of the value, then the field is not required.

Reply:
Sorry my misread. I thought this condition means required.